wallet/udb/txmined.go

```diff
@@ -3520,6 +3520,38 @@ func (s *Store) balanceFullScan(ns, addrmgrNs walletdb.ReadBucket, minConf int32
 			}
 			votingAuthorityAmt := dcrutil.Amount(0)
 			lockedByTicketsAmt := dcrutil.Amount(0)
+			totalInputAmount := dcrutil.Amount(0)
+			for _, txin := range rec.MsgTx.TxIn {
+				rawUnmined := existsRawUnmined(ns, txin.PreviousOutPoint.Hash[:])
+				if rawUnmined != nil {
+					serializedTx := extractRawUnminedTx(rawUnmined)
+					var txInRec TxRecord
+					err = txInRec.MsgTx.Deserialize(bytes.NewReader(serializedTx))
+					if err != nil {
+						return err
+					}
+					for j, txout := range txInRec.MsgTx.TxOut {
```

Comment:
Don't need to loop here, just check that the length is long enough (error otherwise) and then index the slice.